tests/test_encoder.py

```diff
@@ -408,3 +408,159 @@ def test_build_transaction(codec_rpc):
+        ),
+        (
+            FunctionRecipient.ROUTER,
+            Web3.to_checksum_address('0x0000000000000000000000000000000000000000'),
+            Wei(2),
+            Web3.to_checksum_address('0x0000000000000000000000000000000000000002'),
+            HexStr('0x24856bc300000000000000000000000000000000000000000000000000000000000000400000000000000000000000'
+                   '000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000'
+                   '000000000102000000000000000000000000000000000000000000000000000000000000000000000000000000000000'
+                   '000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000'
+                   '000000002000000000000000000000000000000000000000000000000000000000000000600000000000000000000000'
+                   '000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000'
+                   '00000000020000000000000000000000000000000000000000000000000000000000000002')
+        )
+    )
+)
+def test_permit2_transfer_from(
+        codec,
+        function_recipient_input,
+        token_address_input,
+        amount_input,
+        custom_recipient_input,
+        expected_result
+):
+    result = (
+        codec
+        .encode
+        .chain()
+        .permit2_transfer_from(
+            function_recipient=function_recipient_input,
+            token_address=token_address_input,
+            amount=amount_input,
+            custom_recipient=custom_recipient_input
+        )
+        .build()
+    )
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    'trx_speed_input, priority_fee_input, max_fee_per_gas_input, max_fee_per_gas_limit_input, err_msg',
+    (
+        (
+            None,
+            None,
+            Wei(2),
+            None,
+            'Either trx_speed or both priority_fee and max_fee_per_gas must be set.'
+        ),
+        (
+            None,
+            Wei(2),
+            None,
+            None,
+            'Either trx_speed or both priority_fee and max_fee_per_gas must be set.'
+        ),
+        (
+            TransactionSpeed.FAST,
+            Wei(1),
+            Wei(2),
+            None,
+            "priority_fee and max_fee_per_gas can't be set with trx_speed"
+        ),
+        (
+            TransactionSpeed.FAST,
+            None,
+            Wei(1),
+            None,
+            "priority_fee and max_fee_per_gas can't be set with trx_speed"
+        ),
+        (
+            TransactionSpeed.FAST,
+            None,
+            None,
+            0,
+            'Computed max_fee_per_gas is greater than max_fee_per_gas_limit. Either provide max_fee_per_gas, '
+            'increase max_fee_per_gas_limit or wait for less strained conditions'
+        ),
+    )
+)
+def test_build_transaction_value_err(
```

Comment:
This one does not impact the coverage and is a bit redundant with parts of `test_build_transaction()`, but is cleaner and more comprehensive, so I'm happy to keep it.